expression/builtin_time.go

```diff
@@ -5540,6 +5555,15 @@ func (b *builtinQuarterSig) evalInt(row chunk.Row) (int64, bool, error) {
 	}
 
 	if date.IsZero() {
+		// MySQL compatibility, #11203
+		// 0 | 0.0 should be converted to 0 value (not null)
+		n, isNull, err := b.args[0].EvalInt(b.ctx, row)
```

Comment:
`args[0]` is of type `ETDatetime` or `ETTimestamp`, we should not call `evalInt` on it.

Reply:
Fixed